ITL-2 Implement Todo List

EmanuelRusuu opened the pull request:
- ITL-11 Create the Placeholder image when the user is in the home page and no todos added to todo list
- ITL-12 Mobile friendly looking
- ITL-13 Make the todo list look

EmanuelRusuu committed:
ITL-11 Placeholder image when the user has no added todos

- deleted / replaced images to get started with mobile-first design
- started the design for mobile (Iphone 11 Pro)
- deleted TodoForm.vue, it was not needed anymore/for now

EmanuelRusuu committed:
ITL-12 Mobile friendly looking

- responsive on any device

EmanuelRusuu committed:
ITL-13 Make the todo list look

- I did the todo list design as on figma
- It is responsive and looks good for any device

komaromiGyorgy commented on todoapp/src/components/TodoList.vue:
```diff
@@ -1,29 +1,43 @@
 function addTodo() {
-  const newTodo: TextString = {
-    text: inputContent.value
+  const currentDate = new Date()
```
please use a library for date formatting like dayjs or moment

komaromiGyorgy commented on todoapp/src/components/TodoList.vue:
```diff
@@ -1,29 +1,43 @@
+const todos = ref<TodoTypes[]>([])
+const isMobile = ref(false)
+
+function handleResize() {
```
you can use vueuse core which provides these kind of utilitiies

komaromiGyorgy commented on todoapp/src/components/Header.vue:
```diff
@@ -27,28 +30,38 @@ function handleAdd() {
   background-color: var(--add-todo-btn);
 }
+
+@media screen and (min-width: 650px) {
```
please use a more [common breakpoint ](https://testsigma.com/blog/css-breakpoints/#:~:text=While%20there%20is%20no%20hard,1920px%20(for%20desktop%20computers).)

komaromiGyorgy commented on todoapp/src/components/Todo.vue:
```diff
@@ -1,40 +1,210 @@
-</script>
 
+function statusState(todo: TodoTypes) {
+  if (todo.status === false) {
```
please name functions after action so this would be something like toggleTodoState()

Also you can just simply write: 
`
todo.status = !todo.status
`
Which will basically do the same thing

roli-wg commented on todoapp/src/components/Header.vue:
```diff
@@ -15,7 +18,7 @@ function handleAdd() {
     <h1 class="header-title">To do list</h1>
     <button @click="handleAdd" class="add-todo-btn">
-      <img :src="plus" />
+      <img :src="isMobile ? addTodo : addTodoDesktop" />
```
This should act like a real button so please think about active state, disabled state and hover effect. Also you don't need to export the image from the design you can use an icon library for the "+" icon like [heroicons](https://heroicons.com/) and the green circle you will achieve using css

roli-wg commented on todoapp/src/components/Todo.vue:
```diff
@@ -1,40 +1,210 @@
-
 <script setup lang="ts">
-import type { TextString } from '../types/text'
+import type { TodoTypes } from '../types/text'
```
Commonly the type names are singular and starts whit a capital letter so IMO a better name for this could be `Todo`

EmanuelRusuu replied:
I will rename it TodoType if it`s ok, I already have a component named "Todo".

roli-wg commented on todoapp/src/components/Todo.vue:
```diff
@@ -1,40 +1,210 @@
+function handleImportance(priority: number) {
+  if (priority === 0) {
+    return 'low'
+  } else if (priority === 1) {
```
First of all you don't need to use else if you already have a return in the if but even a better solution could be if you has an object
`const priority = {
    0: 'low',
    1: 'medium',
    2: 'high'
}`
and than the function will be just returning the index from the object
`function handleImportance(index: keyof typeof priority) {
    return priority[index]
}`

roli-wg commented on todoapp/src/components/Todo.vue:
```diff
@@ -1,40 +1,210 @@
+      </div>
+      <div class="todo-imp-status-rem">
+        <div class="todo-importance-container-desktop">
+          <p v-if="todo.priority === 0" class="todo-importance-desktop low">Low</p>
```
Can you use the same `handleImportance(todo.priority)` function here as well?

roli-wg commented on todoapp/src/components/Todo.vue:
```diff
@@ -1,40 +1,210 @@
+      </div>
+    </div>
 
+    <div class="todo-desktop" v-else>
```
Please try to have a single todo card for mobile and desktop without duplicating the code.
Try to resolve as much as you can from css and use if else only for part where is truly necessary

roli-wg commented on todoapp/src/components/Todos.vue:
```diff
@@ -1,26 +1,37 @@
-    <p v-else>You have no tasks</p>
-    <TodoUl :todos="todos" @remove-todo="removeTodo" />
+    <TodoUl :is-mobile="isMobile" :todos="todos" />
+    <img :src="isMobile ? NoTodosMobile : NoTodos" v-if="todos.length < 1" />
```
The svg for no todo is the same for desktop and mobil so please use only one svg and create a separate vue component for this empty state which has an img tag with the svg and a p tag for the text below and the only difference between the mobile and desktop version will be the size of image and the font size

EmanuelRusuu committed:
Solved git changes request

- replaced the new Date() with "moment" library,
- added 'created_at' for desktop view as well,
- removed unnecessary screen size retrieval,
- used common breakpoints,
- renamed function "statusState" to "toggleTodoState",
- replaced entire if/else code block with simply returning "!value",
- instead of an image for "add" button and for the empty state of the todos, I used a button with a "plus" icon, and a image  with a paragraph for when the user has no todos,
- renamed the "todo" types file,
- instead of using an entire if/else logic, went for a variable priority where for a value between 0, 1 and 2, we get a string so we can use it in multiple places,
- from two todo cards, one for mobile and one for desktop, went for only one, and solved it with CSS only.

EmanuelRusuu committed:
Removed unnecessary component

- removed TodoUl components
- style changes
